app/controllers/automated_tests_controller.rb

```diff
@@ -220,4 +222,17 @@ def required_params
   def assignment_params
     params.require(:assignment).permit(*required_params)
   end
+
+  def find_or_create_test_student(assignment_id)
+    test_group = Group.find_or_create_by(group_name: 'test_student_group')
+    user = TestStudent.find_or_create_by(user_name: 'test_student',
+                                         first_name: 'Test', last_name: 'Student', hidden: true)
+    test_grouping = Grouping.find_by(group_id: test_group.id, assessment_id: assignment_id)
```

Comment:
You can use `find_or_create_by` here as well and you can use it to find or create the student membership as well.